controllers/datamover.go

```diff
@@ -290,14 +294,82 @@ func (r *DPAReconciler) createResticSecretsPerBSL(dpa *oadpv1alpha1.DataProtecti
 				)
 			}
 		}
-		// TODO: Azure & GCP
+	case AzureProvider:
+		{
+			secretName, secretKey := r.getSecretNameAndKey(&bsl.Spec, oadpv1alpha1.DefaultPluginMicrosoftAzure)
+			bslSecret, err := r.getProviderSecret(secretName)
+			if err != nil {
+				return nil, err
+			}
+
+			// parse the secret and get azure storage account key
+			azcreds, err := r.parseAzureSecret(bslSecret, secretKey)
+			if err != nil {
+				r.Log.Info(fmt.Sprintf("Error parsing provider secret %s for backupstoragelocation %s/%s", secretName, bsl.Namespace, bsl.Name))
+				return nil, err
+			}
+
+			// check for AZURE_ACCOUNT_NAME from BSL
+			if len(bsl.Spec.Config["storageAccount"]) == 0 {
+				return nil, errors.New("no storageAccount value present in backupstoragelocation config")
+			}
+
+			// check for AZURE_STORAGE_ACCOUNT_ACCESS_KEY value
+			if len(bsl.Spec.Config["storageAccountKeyEnvVar"]) != 0 {
+				if azcreds.strorageAccountKey == "" {
+					r.Log.Info("Expecting storageAccountKeyEnvVar value set present in the credentials")
+					return nil, errors.New("no strorageAccountKey value present in credentials file")
+				}
+			}
+
+			accountName := bsl.Spec.Config["storageAccount"]
+			accountKey := azcreds.strorageAccountKey
+
+			// lets construct the repo URL
+			repo := "azure:" + bsl.Spec.ObjectStorage.Bucket + ":"
+
+			// We are done with checks no lets create the azure dm secret
+			rsecret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("%s-volsync-restic", bsl.Name),
+					Namespace: bsl.Namespace,
+					Labels: map[string]string{
+						oadpv1alpha1.OadpBSLnameLabel:     bsl.Name,
+						oadpv1alpha1.OadpOperatorLabel:    "True",
+						oadpv1alpha1.OadpBSLProviderLabel: bsl.Spec.Provider,
+					},
+				},
+			}
+
+			op, err := controllerutil.CreateOrUpdate(r.Context, r.Client, rsecret, func() error {
+
+				err := controllerutil.SetControllerReference(dpa, rsecret, r.Scheme)
+				if err != nil {
+					return err
+				}
+
+				return r.buildDataMoverResticSecretForAzure(rsecret, accountName, accountKey, pass, repo)
+			})
+
+			if err != nil {
+				return nil, err
+			}
+			if op == controllerutil.OperationResultCreated || op == controllerutil.OperationResultUpdated {
+				r.EventRecorder.Event(rsecret,
+					corev1.EventTypeNormal,
+					"ResticSecretReconciled",
+					fmt.Sprintf("%s restic secret %s", op, rsecret.Name),
+				)
+			}
+
+		}
 	}
```

Comment:
- Add a case for empty provider
- Add a default case for unsupported providers

Reply:
@kaovilai we do not support empty provider or any other provider other than aws/azure/gcp, its already validated in bsl.go https://github.com/openshift/oadp-operator/blob/master/controllers/bsl.go#L61